docs/resources/acl_auth_method.md

```diff
@@ -54,8 +80,8 @@ The following arguments are supported:
 * `type` - (Required) The type of the ACL auth method.
 * `display_name` - (Optional) An optional name to use instead of the name
   attribute when displaying information about this auth method.
-* `max_token_ttl` - (Optional) The maximum life of any token created by this
-  auth method.
+* `max_token_ttl` - The maximum life of any token created by this
+  auth method. (Required) for OIDC auth method
```

Comment:
```suggestion
* `max_token_ttl` - (Optional)The maximum life of any token created by this
  auth method. **This attribute is required for OIDC auth method.**
```

Reply:
is this true? optional for everything else, but only required for oidc?

Reply:
I don't know about the other auth methods, I only know for sure is required for OIDC

Reply:
@im2nguyen 
I just tested all other 3 Auth methods and TF didn't error on missing max_token_ttl. so It's optional for all except OIDC

Reply:
one small note, in this context it's an argument not an attribute

Reply:
Confirming that `max_token_ttl` is required for OIDC. <https://developer.hashicorp.com/consul/api-docs/acl/auth-methods#maxtokenttl>

> [MaxTokenTTL](https://developer.hashicorp.com/consul/api-docs/acl/auth-methods#maxtokenttl) (duration: 0s) - This specifies the maximum life of any token created by this auth method.…This value must be no smaller than 1 minute and no longer than 24 hours. Added in Consul 1.8.0.
>
> **This must be set to a nonzero value for type=oidc.**

Reply:
Thanks for the confirmation!